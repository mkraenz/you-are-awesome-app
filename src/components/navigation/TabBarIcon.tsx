import { FontAwesome } from "@expo/vector-icons";
import React, { FC } from "react";
import { StyleSheet } from "react-native";
import { useTheme } from "react-native-paper";

const styles = StyleSheet.create({
    icon: {
        marginBottom: -3,
    },
});

interface Props {
    name: string;
    focused: boolean;
}

const TabBarIcon: FC<Props> = ({ name, focused }) => {
    const { colors } = useTheme();

    return (
        <FontAwesome
            name={name}
            size={26}
            style={styles.icon}
            color={focused ? colors.accent : colors.disabled}
        />
    );
};

export default TabBarIcon;
