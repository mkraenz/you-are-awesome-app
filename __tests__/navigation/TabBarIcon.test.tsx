import React from "react";
import "react-native";
import { DefaultTheme, Provider as PaperProvider } from "react-native-paper";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import TabBarIcon from "../../src/components/navigation/TabBarIcon";

jest.mock("@expo/vector-icons", () => ({ FontAwesome: "Fontawesome" }));

it("renders correctly", () => {
    const tree = renderer
        .create(
            <PaperProvider theme={DefaultTheme}>
                <TabBarIcon name="my-name" focused={false} />
            </PaperProvider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it("renders correctly with focus", () => {
    const tree = renderer
        .create(
            <PaperProvider theme={DefaultTheme}>
                <TabBarIcon name="my-name" focused />
            </PaperProvider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
