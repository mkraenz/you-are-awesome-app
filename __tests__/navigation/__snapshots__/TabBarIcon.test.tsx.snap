// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Fontawesome
    color="rgba(0, 0, 0, 0.26)"
    name="my-name"
    size={26}
    style={
      Object {
        "marginBottom": -3,
      }
    }
  />
</View>
`;

exports[`renders correctly with focus 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Fontawesome
    color="#03dac4"
    name="my-name"
    size={26}
    style={
      Object {
        "marginBottom": -3,
      }
    }
  />
</View>
`;
